### PR TITLE
send notifications as email

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

The notifications shown in the WebUI are not so usefull for people who not login very often.
Send them also as email if email notifications in the user preferences is enabled.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- not needed

- [x] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Links

Tracks https://github.com/SUSE/spacewalk/pull/19751

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by mcalmer** [comment]
CI: https://ci.nue.suse.com/view/Manager/view/Uyuni-PRs/job/uyuni-prs-ci-tests/1396/

**Comment by mcalmer** [comment]
Missing: Notifications created from the python stack (Bootstrap repo generation) is entered directly into the DB and do not send an email.
Also "Channel Sync Failed" on reposync errors do not send mails. Need investigation

**Review by lucidd — APPROVED** [review]
besides what thomas already mentioned looks good.

File: python/uyuni/common/notificationUtils.py [hunk 1]
```diff
@@ -23,49 +33,9 @@ def store(self):
...
-        """)
+        return self._create_bootstrap_repo_failed_notification()
 
-        for user in affected_user:
-            j.execute(uid=user['id'], mid=mid)
-        rhnSQL.commit()
+    def _create_bootstrap_repo_failed_notification(self):
+        client = xmlrpc_client.Server(TASKOMATIC_XMLRPC_URL)
+        log_debug(2, "Calling createBootstrapRepoFailedNotification({0}, {1})".format(self.identifier, self.details))
+        client.tasko.createBootstrapRepoFailedNotification(self.identifier, self.details)
```
Comment (cbosdo) [inline python/uyuni/common/notificationUtils.py:41]: Missing `return`  here

**Review by cbosdo — CHANGES_REQUESTED** [review]
Spot a problem in the python code. It would also be nice to address the reported code smells or at least mark them as wontfix.
https://sonarcloud.io/project/issues?id=uyuni-project_uyuni&pullRequest=5512&resolved=false&types=CODE_SMELL

File: python/uyuni/common/notificationUtils.py (hunk 1 above)
Comment (mcalmer) [reply python/uyuni/common/notificationUtils.py]: fixed

**Review by cbosdo — APPROVED** [review]

**Pull request merged**